### ISSUE TITLE
Helper function to extract images and generate different sizes

### ISSUE THREAD

**Issue by acrosman** [tamw-wnet/PBS_Media_Manager_Client#21]
Both WETA and SCETV need to be able to pull images from shows and episodes and ideally resize them using the PBS image handling service. For example see Latest Episodes section of: https://www.scetv.org/program/reel-south

It seems likely that others may find it useful to be able to do similar things so I figured before we generalize what I have for SCETV into the Drupal module we should check to see if it makes sense to add it to the main client library instead.

Basically I'm proposing adding a helper function that would take an asset with a image, and return a specific image profile. The current functions I use are in the form:
`extract_image($data, $image_type, $width, $height)` where `$image_type` are image profiles like `Mezzanine`
and 
`image_path($baseURL, $width, $height, $type = '')`  where `$type` is the image type itself like `jpg`.

**Comment by tamw-wnet** [comment]
There's a lot of complexity with the image profile stuff, beyond resizing -- there's 'poster' and 'logo' profiles that are radically different than the various resizes of the Mezzanine.

I do like the idea of better integrating pulling the image data that we get from the objects and making it available based on the known profiles.

One thing that we do in our application is just use the PBS.org urls, including the resizer urls.

Lemme suggest that we simplify the call a little to something like
`get_image($object_id, $profile)`
where $object_id is the asset/episode/whatever content id and $profile is one of the predefined profiles.  That would return the PBS.org hosted URL for the image.
Since height/width can always be modified after the fact by just adding .crop.385x217.jpg or .resize.385x217.jpg to the url, I'd leave that out.

**Comment by acrosman** [comment]
Yeah, that complexity is some of what I'm trying to abstract out.  It would be pretty simple to separate them more cleanly so that we have a `get_image()` as you describe. The `extract_image()` function I have actually does what you're suggesting, but then passes off to `image_path()` to add the adjustment stuff onto the URL.  Moving that second step out of that function is trivial.  

Since it sounds like you're interested I'll put together a pull request with this in the next couple days and we can refine it from there.
